### PR TITLE
Make ansible-lint and yamllint use more strict rules.

### PR DESCRIPTION
**Pull request by rjeffman**
This patch modifies configuration of both ansible-lint and yamllint
to check for more rules, resulting in a more strict verification.

The rules left disable are either necessary due to missing Ansible
modules, or might impact modules or roles execution or testing.

### CODE REVIEW

**Comment by rjeffman** [comment]
@t-woerner, please, can you review this PR?

**Comment by rjeffman** [comment]
PR rebased.

File: .yamllint [hunk 1]
```diff
@@ -16,13 +16,8 @@ rules:
   truthy:
     allowed-values: ["yes", "no", "true", "false", "True", "False"]
     level: error
+  # Ansible line length is 160.
```
Comment (t-woerner) [inline .yamllint:19]: Why is ling length set to 160?

Comment (rjeffman) [reply .yamllint:19]: IIRC this is the default for ansible-lint.

Comment (rjeffman) [reply .yamllint]: If not set, yamllint will consider a line with more than 80 characters an error. In older versions of ansible-lint, IIRC, it warned about lines longer that 160 characters (it seems that with the most recent ones, this is not as issue anymore).

That's why I used 160 as the maximum line length. Seemed to me a good compromise in readability and ease to write playbooks. With this value, almost only certificate data need to be formatted to fit in the maximum size.

File: .ansible-lint [hunk 2]
```diff
@@ -8,23 +9,22 @@ exclude_paths:
...
+  - meta/runtime.yml
+ 
+kinds:
+  - playbook: '**/tests/**/test_*.yml'
+  - playbook: '**/playbooks/**/*.yml'
 
 parseable: true
 
-quiet: false
+quiet: yes
```
Comment (t-woerner) [inline .ansible-lint:20]: What is the result of using quiet here?

File: .ansible-lint [hunk 3]
```diff
@@ -8,23 +9,22 @@ exclude_paths:
...
   - '301'  # Commands should not change things if nothing needs doing'
-  - '305'  # Use shell only when shell functionality is required'
-  - '306'  # Shells that use pipes should set the pipefail option'
-  - '502'  # All tasks should be named
-  - '505'  # Referenced missing file
-
+  - '305'  # Use shell only when shell functionality is required
+  - '306'  # risky-shell-pipe
+  - yaml   # yamllint should be executed separately.
+  - experimental  # Ignore rules that resulted in false positives
```
Comment (t-woerner) [inline .ansible-lint:27]: What is `experimental` exactly?

File: playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml [hunk 4]
```diff
@@ -5,8 +5,8 @@
   gather_facts: false
 
   tasks:
-  # Ensure DNS zone is present
-  - ipadnsforwardzone:
+  - name: Ensure DNS forward zone is present with forward port
```
Comment (t-woerner) [inline playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml:8]: Is the port really important to mention here?

File: playbooks/dnsrecord/ensure-dnsrecord-is-absent.yml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-- name: Test DNS Record is absent.
+- name: Playbook to manage DNS records.
   hosts: ipaserver
   become: true
   gather_facts: false
 
   tasks:
-  # Ensure that dns record is absent
-  - ipadnsrecord:
+  - name: Ensure that dns record is absent
```
Comment (t-woerner) [inline playbooks/dnsrecord/ensure-dnsrecord-is-absent.yml:8]: I think something like `Ensure that dns localhost AAAA record is absent` would be better.

File: .ansible-lint (hunk 2 above)
Comment (rjeffman) [reply .ansible-lint:20]: It actually hides some info that we might want. It should be "false".

File: playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml (hunk 4 above)
Comment (rjeffman) [reply playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml:8]: Not really. It is here just to differentiate from the other playbook.

Do you think it is better to remove the mention to the port?

File: .ansible-lint (hunk 3 above)
Comment (rjeffman) [reply .ansible-lint]: `experimental` varies from release to release. It used to give lots of false positive, but is much more stable in the latest versions.

I enabled `experimental` and fixed the warnings for `ansible-lint 5.0.7 using ansible 2.9.25`.

File: .ansible-lint (hunk 2 above)
Comment (rjeffman) [reply .ansible-lint]: Fixed.

File: playbooks/dnsrecord/ensure-dnsrecord-is-absent.yml (hunk 5 above)
Comment (rjeffman) [reply playbooks/dnsrecord/ensure-dnsrecord-is-absent.yml]: Fixed.

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -7,7 +7,9 @@ repos:
     always_run: false
     pass_filenames: true
     files: \.(yaml|yml)$
-    entry: env ANSIBLE_LIBRARY=./plugins/modules ANSIBLE_MODULE_UTILS=./plugins/module_utils ANSIBLE_DOC_FRAGMENT_PLUGINS=./plugins/doc_fragments ansible-lint --force-color
+    exclude: /env[^/]*.(yaml|yml)$
```
Comment (t-woerner) [inline .pre-commit-config.yaml:10]: What is this excluding exactly?

File: playbooks/host/host-member-managedby_host-absent.yml [hunk 7]
```diff
@@ -4,9 +4,10 @@
   become: true
 
   tasks:
-  - ipahost:
+  - name: Ensure host member managedby_host 'server.example.com' is absent
+    ipahost:
       ipaadmin_password: SomeADMINpassword
-      name: host01.exmaple.com
+      name: host01.example.com
       managedby_host: server.exmaple.com
```
Comment (t-woerner) [inline playbooks/host/host-member-managedby_host-absent.yml:11]: Please also fix this "exmaple".

Comment (t-woerner) [reply playbooks/host/host-member-managedby_host-absent.yml:11]: We have a lot of "exmaple" in the playbooks folder...

Comment (t-woerner) [reply playbooks/host/host-member-managedby_host-absent.yml:11]: Here is a new PR to fix all the exmaples: #649

File: utils/lint_check.sh [hunk 8]
```diff
@@ -1,29 +1,35 @@
...
+echo "Running 'pydocstyle'..."
 pydocstyle .
+echo "Running 'pylint'..."
 pylint plugins
 
-ANSIBLE_LIBRARY=${ANSIBLE_LIBRARY:-"${topdir}/plugins/modules"}
+ANSIBLE_LIBRARY="${ANSIBLE_LIBRARY:-${topdir}/plugins/modules}"
 ANSIBLE_MODULE_UTILS=${ANSIBLE_MODULE_UTILS:-"${topdir}/plugins/module_utils"}
 
 export ANSIBLE_LIBRARY ANSIBLE_MODULE_UTILS
```
Comment (t-woerner) [inline utils/lint_check.sh:18]: ANSIBLE_DOC_FRAGMENT_PLUGINS needs to be set also.

File: playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml (hunk 4 above)
Comment (t-woerner) [reply playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml:8]: A "forward port" is a firewall thing for me.

Comment (rjeffman) [reply playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml:8]: Makes sense. I'll rephrase it.

File: utils/lint_check.sh (hunk 8 above)
Comment (rjeffman) [reply utils/lint_check.sh:18]: Fixed.

File: playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml (hunk 4 above)
Comment (rjeffman) [reply playbooks/dnsforwardzone/ensure-dnsforwardzone-with-forwarder-port.yml:8]: (Hopefully) fixed.

**Review by t-woerner — APPROVED** [review]
LGTM

**Pull request merged**